qiskit_cold_atom/fermions/ffsim_backend.py

```diff
@@ -0,0 +1,388 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fermionic simulator backend that uses ffsim."""
+
+from __future__ import annotations
+
+import datetime
+import time
+import uuid
+import warnings
+from collections import Counter
+from typing import Any, Dict, List, Optional, Union
+
+import ffsim  # pylint: disable=import-error
+import numpy as np
+import scipy.linalg
+from qiskit import QuantumCircuit
+from qiskit.providers import Options
+from qiskit.providers.models import BackendConfiguration
+from qiskit.result import Result
+from qiskit_aer import AerJob
+from qiskit_nature.operators.second_quantization import FermionicOp
+from scipy.sparse.linalg import expm_multiply
+
+from qiskit_cold_atom.circuit_tools import CircuitTools
+from qiskit_cold_atom.fermions.base_fermion_backend import BaseFermionBackend
+from qiskit_cold_atom.fermions.fermion_gate_library import (
+    FermionicGate,
+    Hop,
+    Interaction,
+    LoadFermions,
+    Phase,
+)
+
+
+class FfsimBackend(BaseFermionBackend):
+    """Fermionic simulator backend that uses ffsim.
+
+    This is a specialized simulator backend for fermionic circuits that uses `ffsim`_.
+    It computes the statevector and simulate measurements with vastly improved efficiency
+    compared with the :class:`~.FermionSimulator` backend. However, unlike
+    :class:`~.FermionSimulator`,
+    it does not compute the full unitary of a circuit.
+
+    This backend is not supported on Windows, and a special ``pip`` command is needed
+    to install it:
+
+    .. code::
+
+        pip install "qiskit-cold-atom[ffsim]"
+
+    .. _ffsim: https://github.com/qiskit-community/ffsim
+    """
+
+    _DEFAULT_CONFIGURATION = {
+        "backend_name": "ffsim_simulator",
+        "backend_version": "0.0.1",
+        "n_qubits": 100,
+        "basis_gates": None,
+        "gates": [],
+        "local": False,
+        "simulator": True,
+        "conditional": False,
+        "open_pulse": False,
+        "memory": True,
+        "max_shots": 1e6,
+        "coupling_map": None,
+        "description": "ffsim simulator for fermionic circuits. Instead of qubits, each wire represents"
+        " a single fermionic mode",
+        "supported_instructions": None,
```

Comment:
Would be great to add here the gates that are allowed in terms of strings. This makes it accessible to the user via backend.configuration().supported_instructions. Then it is also clear that the X,Y,Z gate are not allowed.

Reply:
I copied this from the other simulator backend. I'm not sure what it should be set to because arbitrary FermionicGates are supported, but those don't have a name. I also noticed that other simulators, like in Qiskit Aer, don't set this property. Perhaps we should just leave it as is?

Reply:
FYI I added supported for the FR{X,Y,Z}Gates.